modules/json/tests/test_format_json.c

```diff
@@ -224,10 +224,17 @@ Test(format_json, test_v40_value_pairs_yields_typed_values)
                          "{\"FACILITY_NUM\":19}");
 
   /* RFC8259 number sanitization */
-  assert_template_format("$(format-json num=int(00014))", "{\"num\":14}");
+  assert_template_format("$(format-json num=int(+00014))", "{\"num\":12}");
```

Comment:
good point. hmm, maybe if we didn't support octal, only hexadecimal? the 0 prefix is not a good enough indicator and data is often just padded this way.

Reply:
I think supporting octal would be great.
What do you think about adding a template function or an alternative type-cast that could get rid of those zeros easily?

Reply:
db-parser() is also limited to hex numbers, octals are not supported there either.

If I imagine log messages with numbers in them, they are more likely to contain zero padded decimals than octals.

If we were to support octals I'd do that as a separate feature, e.g. as a template function that takes an octal and converts it to decimal.

Assuming zero padded numbers are octals would do more harm than good, I think. 

The 0x prefix is clear, and cannot be interpreted as a decimal number, so that's just smart. Turning a 0 padded number to something else would cause surprises.

Reply:
Okay, sounds great to me :)

Reply:
just pushed a version that should only "understand" hex numbers, just as db-parser()

if a use-case comes up for octal numbers we could implement either as a template function or as a type hint. Although "octal" number is not really a type.